EIPS/eip-draft_disable_selfdestruct.md

```diff
@@ -0,0 +1,63 @@
+Disable SELFDESTRUCT
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+Make `SELFDESTRUCT` EVM operation no-op (effectively disable). Contracts will not be able to delete themselves.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+Althouth `SELFDESTRUCT` originally came with EVM to help clean up the state, we learnt that in practice it did not achieve this objective on
+sufficient scale, and is on balance a burden of extra complexity and unintended effects (GasToken2 and polymorphic contracts).
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+Operation `SELFDESTRUCT` (formerly `SUICIDE`) was in the EVM from the beginning, and its purpose was to incentivise clearing the state, by giving the
+caller gas refund. In practice, this incentivisation turned out to be very limited, and the designed purpose was not archieved. However, `SELFDESTRUCT`
+brings significant amount of complexity to the EVM. It is responsible for some of the most arcane edge cases in the EVM semantics.
+It is also used as a vehicle to run an efficient arbitrage of gas prices (GasToken2), which
+ironically lead to the increase use of the state. Introduction of `CREATE2` opcode in Constantinople upgrade created a new phenomenom of
+polymorthic contracts, i.e. contracts that can change their bytecode over time, while residing on the same address. Polymorphic contracts are limited
+in their use, because changing the bytecode via `SELFDESTRUCT` + `CREATE2` clears all the contract storage, making contract lose all its data,
+and making it unsuitable to replace Proxy Pattern as a technique for upgradable contracts. Removing the effect of `SELFDESTRUCT`
+will reduce complexity of EVM going forward, disable GasToken2 (but not GasToken1, which is based on storage clearing refunds), and make
+polymorthic contracts impossible.
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+After certain block number, the semantics of `SELFDESTRUCT` becomes the same as the combination of `POP`, followed by transferring remaining ETH
+to the address popped from the stack, followed by `STOP`. Gas cost is the same as the gas cost of non-zero value transfer, which is 9000 gas. No gas refund is given.
```

Comment:
I don't think there's any point introducing new pricing here, if the specification is that `SELFDESTRUCT` is replaced with `CALL(<addr>, <value>, gas=0, ..`) and `STOP`, then just state that and apply the gas rules of those two.

Reply:
This opertation should be cheaper than CALL because it doesn't need to load code or create an execution context.

Reply:
> because it doesn't need to load code or create an execution context.

This EIP is changing the semantics of SELFDESTRUCT. From the current description it is not clear whether it turns into a non-executing transfer or not.

Reply:
Hm, turning it into an executing transfer would be a nice EVM unification/simplification. `SELFDESTRUCT` is currently one of the "sneakiest" ways to add funds to an address; a capability that people often forget about.

Reply:
@carver Developers only "forget" about it because of [deceptive Solidity syntax](https://github.com/ethereum/solidity/issues/7727). Nobody should assume their contract is "nonpayable". This does not change that because miners can still break that assumption via coinbase.